OpenRA.Game/Traits/Player/FrozenActorLayer.cs

```diff
@@ -179,7 +186,27 @@ public void Tick(Actor self)
 		public virtual IEnumerable<IRenderable> Render(Actor self, WorldRenderer wr)
 		{
 			return world.ScreenMap.FrozenActorsInBox(owner, wr.Viewport.TopLeft, wr.Viewport.BottomRight)
-				.Where(f => f.Visible)
+				.Where(f =>
```

Comment:
OK. I made that to fix the bug #1 described above. The frozen actor of the building has to be rendered at least once so that it is cached and is shown even when the building is not there anymore. This code makes sure that every frozen actor is rendered once even if it's under disruption field and then one tick passes while it is rendered. Because tick and render frequencies are different. This makes the renderables cached so that they can be shown again when the building is destroyed and disruption field is gone.

At least this is my understanding of how the engine works. Is that correct?